test/test-apimatic-core/union_type_tests/any_of_test.rb

```diff
@@ -0,0 +1,2461 @@
+require 'minitest/autorun'
+require 'apimatic_core'
+
+require_relative '../../test-helper/models/morning'
+require_relative '../../test-helper/models/evening'
+require_relative '../../test-helper/models/month_number_enum'
+require_relative '../../test-helper/models/month_name_enum'
+
+class AnyOfTest < Minitest::Test
+  include CoreLibrary, TestComponent
+
+  def setup
+  end
+
+  def teardown
+  end
+
+  # === Simple cases ===
+
+  def test_valid_string_in_any_of
+    _any_of = AnyOf.new([LeafType.new(String), LeafType.new(Integer)])
+    _any_of.validate('string')
+    assert _any_of.is_valid
+  end
+
+  def test_valid_integer_in_any_of
+    _any_of = AnyOf.new([LeafType.new(String), LeafType.new(Integer)])
+    _any_of.validate(4)
+    assert _any_of.is_valid
+  end
+
+  def test_valid_float_in_any_of
+    _any_of = AnyOf.new(
+      [
+        LeafType.new(String),
+        LeafType.new(Float)
+      ]
+    )
+    _any_of.validate(4.0)
+    assert _any_of.is_valid
+  end
+
+  def test_valid_integer_float_in_any_of
+    _any_of = AnyOf.new([LeafType.new(Integer), LeafType.new(Float)])
+    _any_of.validate(4)
+    assert _any_of.is_valid
+  end
+
+  def test_valid_boolean_class_float_in_any_of
+    _any_of = AnyOf.new([LeafType.new(Integer), LeafType.new(TrueClass), LeafType.new(FalseClass)])
+    _any_of.validate(false)
+    assert _any_of.is_valid
+  end
+
+  def test_valid_multiple_types_in_any_of
+    _any_of = AnyOf.new(
+      [
+        LeafType.new(Integer),
+        LeafType.new(Float),
+        LeafType.new(String)
+      ]
+    )
+    _any_of.validate(4)
+    assert _any_of.is_valid
+  end
+
+  def test_valid_both_string_any_of
+    _any_of = AnyOf.new(
+      [
+        LeafType.new(String),
+        LeafType.new(String)
+      ]
+    )
+    _any_of.validate('string')
+    assert _any_of.is_valid
+  end
+
+  def test_invalid_any_of
+    _any_of = AnyOf.new(
+      [
+        LeafType.new(String),
+        LeafType.new(String)
+      ]
+    )
+    assert_raises AnyOfValidationException do
```

Comment:
Blocker: The assertion against the error messages that the exception has thrown is missing. I'd recommend to see the tests in python core library.

Reply:
done